RFC0032.md

```diff
@@ -0,0 +1,112 @@
+
+* Outputting tagged log messages to stdout or stderr of the application, or a connected tool. Where supported, an alternative output stream (possibly directed to a dedicated log file) may be specified. Messages may be tagged (via the PMIX\_LOG\_TAG\_OUTPUT} attribute) as flowing via the PMIx\_Log API to differentiate them from the application's normal output. In addition, messages may be timestamped (PMIX\_LOG\_TIMESTAMP\_OUTPUT) or output in XML format (PMIX\_LOG\_XML\_OUTPUT)
+
+* Storing status updates in the job record using the PMIX\_LOG\_JOB\_RECORD attribute. Resource managers nearly always maintain a record of the jobs they schedule and execute. This includes information on the time spent waiting for allocation, priority of the request, identity of the requestor, name/path of the executable and/or job script, etc. Historically, users have had to record status information on their application (e.g., computational progress) in files which are subsequently stored in persistent storage. PMIx\_Log offers the option (where supported) of injecting such status reports directly into the job record, thus providing a single, time sequential record of the job's execution.
+
+  Note that system libraries can also use this feature to record job-affecting events (e.g., network failures) that might have impacted the application during its execution, perhaps linking them to more detailed information stored in a RAS database.
+
+* Storing state information in a global key-value datastore. The prior use-cases all involve one-way output of data - i.e., the caller can log the data, but cannot later retrieve it. However, there are times when an application, tool, or SMS element may wish to store information in a global key-value datastore that can be searched by external agents, or be retrieved by the caller itself at some later time. For example, an SMS element may wish to store state information in a persistent place for retrieval upon recovery from a failure. Use of the PMIx\_Publish API might, at first glance, appear to meet this need. However, PMIx_Publish is focused on inter/intra-application data exchange - it therefore does not guarantee persistence across (for example) an RM failure.
+
+  Passing the PMIX\_LOG\_GLOBAL\_DATASTORE attribute in a call to PMIx\_Log indicates that the provided data is to be stored in a persistent datastore. Additional attributes can be used to provide direction on the storage strategy - e.g., hot/warm/cold or locality. Note that it is advisable to use the optional flag for storage strategy directives as support for such behaviors is not yet commonplace.
+
```

Comment:
Can I store PMIX_NAMESPACE = 90210 in the  global datastore?   Is there anything that prevents an application from screwing up its own environment?

Reply:
What are these "additional attributes" to provide direction on hot/warm/cold?

Reply:
Per the telecon, I will remove the global datastore references until we better define that capability